Sources/SwiftParser/Parser.swift

```diff
@@ -142,7 +145,12 @@ public struct Parser {
   ///            arena is created automatically, and `input` copied into the
   ///            arena. If non-`nil`, `input` must be within its registered
   ///            source buffer or allocator.
-  public init(_ input: UnsafeBufferPointer<UInt8>, maximumNestingLevel: Int? = nil, arena: ParsingSyntaxArena? = nil) {
+  public init(
+    _ input: UnsafeBufferPointer<UInt8>,
+    maximumNestingLevel: Int? = nil,
+    arena: ParsingSyntaxArena? = nil,
+    parseTransition: IncrementalParseTransition? = nil
```

Comment:
I would put `parseTransition` before `arena` because `arena` is always the last parameter.